interop/fake_grpclb/fake_grpclb.go

```diff
@@ -0,0 +1,178 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Runs a fake grpclb balancer server, can be useful for testing.
+// The name of the service to load balance for and the addresses
+// of that service are provided by command line flags.
+package main
+
+import (
+	"flag"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	durationpb "github.com/golang/protobuf/ptypes/duration"
+	"google.golang.org/grpc"
+	lbpb "google.golang.org/grpc/balancer/grpclb/grpc_lb_v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/alts"
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/testdata"
+)
+
+var (
+	port        = flag.Int("port", 10000, "Port to listen on.")
+	backendIps  = flag.String("backend_addrs", "", "Comma separated list of backend IP/port addresses.")
+	useALTS     = flag.Bool("use_alts", false, "Listen on ALTS credentials.")
+	useTLS      = flag.Bool("use_tls", false, "Listen on TLS credentials, using a test certificate.")
+	shortStream = flag.Bool("short_stream", false, "End the balancer stream immediately after sending the first server list.")
+	serviceName = flag.String("service_name", "UNSET", "Name of the service being load balanced for.")
+)
+
+type backendAddr struct {
+	Ip   []byte
+	Port int32
+}
+
+type loadBalancerServer struct {
+	BackendAddrs []backendAddr
+}
+
+func (l *loadBalancerServer) BalanceLoad(stream lbpb.LoadBalancer_BalanceLoadServer) error {
+	grpclog.Info("Begin handling new BalancerLoad request.")
+	var lbReq *lbpb.LoadBalanceRequest
+	var err error
+	if lbReq, err = stream.Recv(); err != nil {
+		grpclog.Errorf("Error receiving LoadBalanceRequest: %v", err)
+		return err
+	}
+	grpclog.Info("LoadBalancerRequest received.")
+	initialReq := lbReq.GetInitialRequest()
+	if initialReq == nil {
+		grpclog.Info("Expected first request to be an InitialRequest. Got: %v", lbReq)
+		return status.Error(codes.Code(2), "First request not an InitialRequest")
+	}
+	// gRPC clients targetting foo.bar.com:443 can sometimes include the ":443" suffix in
```

Comment:
I remember this was discussed, and is expected.

If we want to remove the port, how about https://golang.org/pkg/net/#SplitHostPort?
Or we require the port number in `serviceName`.

Reply:
let's chat offline. For now, I'm using SplitHostPort